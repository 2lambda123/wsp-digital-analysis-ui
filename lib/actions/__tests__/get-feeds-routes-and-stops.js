/* global describe, expect, it */

import nock from 'nock'

import getFeedsRoutesAndStops from '../get-feeds-routes-and-stops'

describe('actions > getFeedsRoutesAndStops', () => {
  it('should work when fetching via getAllRoutesAndStops and then getUnfetchedRoutes', async () => {
    const mockRoutes = [
      {
        route_id: '1',
        route_short_name: '1'
      }
    ]
    const mockResponse = {
      bundle: [
        {
          feeds: [
            {
              checksum: 'abcd',
              detailRoutes: mockRoutes,
              feed_id: '1',
              routes: mockRoutes,
              stops: [
                {
                  stop_id: '1'
                }
              ]
            }
          ]
        }
      ]
    }

    const nockHost = nock('http://mockHost.com/')
    nockHost.get(/^\/api\/graphql/)
      .reply(200, mockResponse)

    const getResult = getFeedsRoutesAndStops({ bundleId: 1 })

    // expect to receive increment action while making request
    expect(getResult[0]).toMatchSnapshot()

    // perform request
    const fetchResult = await getResult[1]

    // expect to receive decrement action upon fulfillment of request
    expect(fetchResult[0]).toMatchSnapshot()

    // parse response
    const parseResult = await fetchResult[1]

    // expect parse result to be what we desired
    expect(parseResult).toMatchSnapshot()

    // make request again, forcing fetching via getUnfetchedRoutes
    nockHost.get(/^\/api\/graphql/)
      .reply(200, mockResponse)

    const actionParams = {
      bundleId: 1,
      modifications: [
        {
          type: 'reroute',
          routes: ['2']
        }
      ]
    }
    const getResult2 = getFeedsRoutesAndStops(actionParams)

    // perform request
    const fetchResult2 = await getResult2[1]

    // parse response
    const parseResult2 = await fetchResult2[1]

    // expect parse result to be what we desired
    expect(parseResult2).toMatchSnapshot()
  })
})
