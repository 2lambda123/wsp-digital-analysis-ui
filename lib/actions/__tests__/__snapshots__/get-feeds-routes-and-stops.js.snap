exports[`actions > getFeedsRoutesAndStops should work when fetching via getAllRoutesAndStops and then getUnfetchedRoutes 1`] = `
Object {
  "type": "increment outstanding requests",
}
`;

exports[`actions > getFeedsRoutesAndStops should work when fetching via getAllRoutesAndStops and then getUnfetchedRoutes 2`] = `
Object {
  "type": "decrement outstanding requests",
}
`;

exports[`actions > getFeedsRoutesAndStops should work when fetching via getAllRoutesAndStops and then getUnfetchedRoutes 3`] = `
Object {
  "payload": Array [
    Object {
      "checksum": "abcd",
      "id": "1",
      "routes": Array [
        Object {
          "label": "1 ",
          "route_id": "1",
          "route_short_name": "1",
        },
      ],
      "routesById": Object {
        "1": Object {
          "label": "1 ",
          "route_id": "1",
          "route_short_name": "1",
        },
      },
      "stops": Array [
        Object {
          "stop_id": "1",
        },
      ],
      "stopsById": Object {
        "1": Object {
          "stop_id": "1",
        },
      },
    },
  ],
  "type": "set feeds",
}
`;

exports[`actions > getFeedsRoutesAndStops should work when fetching via getAllRoutesAndStops and then getUnfetchedRoutes 4`] = `
Object {
  "payload": Array [
    Object {
      "checksum": "abcd",
      "id": "1",
      "routes": Array [
        Object {
          "label": "1 ",
          "route_id": "1",
          "route_short_name": "1",
        },
      ],
      "routesById": Object {
        "1": Object {
          "label": "1 ",
          "route_id": "1",
          "route_short_name": "1",
        },
      },
      "stops": Array [
        Object {
          "stop_id": "1",
        },
      ],
      "stopsById": Object {
        "1": Object {
          "stop_id": "1",
        },
      },
    },
  ],
  "type": "set feeds",
}
`;
